subprojects/language-native/src/main/java/org/gradle/language/nativeplatform/internal/CompileTaskConfig.java

```diff
@@ -93,7 +93,7 @@ public FileCollection transform(NativeDependencySet original) {
             }
         });
         FileCollectionFactory fileCollectionFactory = ((ProjectInternal) task.getProject()).getServices().get(FileCollectionFactory.class);
-        task.includes(fileCollectionFactory.create(new MinimalFileSet() {
+        task.systemIncludes(fileCollectionFactory.create(new MinimalFileSet() {
```

Comment:
We should add something about this to the release notes

Reply:
Done.